3rdparty/dlmalloc/allocator.c

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/internal/allocator.h>
+
+#define HAVE_MMAP 0
+#define LACKS_UNISTD_H
+#define LACKS_SYS_PARAM_H
+#define LACKS_SYS_TYPES_H
+#define LACKS_TIME_H
+#define MORECORE dlmalloc_sbrk
+#define ABORT oe_abort()
+#define USE_DL_PREFIX
+#define LACKS_STDLIB_H
+#define LACKS_STRING_H
+#define USE_LOCKS 1
+#define fprintf _dlmalloc_stats_fprintf
+#define NO_MALLOC_STATS 1
+
+#pragma GCC diagnostic push
```

Comment:
Could you add a comment here for why this is needed?

Reply:
This may not be needed. Good catch. Will try to remove it and see what happens.

Reply:
Removed diagnostic push